### PR TITLE
Update to 3.1.0 stable from upstream

### LINKED ISSUE #10: Gatekeeper Dashboards should be moved to this project instead of integrating them into gatekeeper
Hi, because of https://github.com/sighupio/fury-kubernetes-monitoring/pull/40, now is possible to decouple dashboards from Gatekeeper, using configmaps with proper annotation.
Because of is Gatekeeper is an optional module of Fury, should deploy dashboards indipendently from Gatekeeper itself

### PR DESCRIPTION
**Pull request by ralgozino**
This PR brings the following changes:
- Updates Gatekeeper to version 3.1.0 the first stable release from upstream. As a side benefit, we can now have the webhook pods in HA.
- Namespaces `kube-system` and `gatekeeper-system` are now exempted for all operations.
- Moved the Grafana dashboard from the monitoring module to this module ( Fixes #10 )
- Added linting step to the pipeline with policeman and done some minor linting corrections.

Please note that the update modifies some labels selector that require a `--force` on the apply. The error you'd get is:

```
The Deployment "gatekeeper-controller-manager" is invalid: spec.selector: Invalid value: v1.LabelSelector{MatchLabels:map[string]string{"control-plane":"controller-manager", "gatekeeper.sh/operation":"webhook", "gatekeeper.sh/system":"yes"}, MatchExpressions:[]v1.LabelSelectorRequirement(nil)}: field is immutable
```

### CODE REVIEW

File: docs/releases/unreleased.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Release notes
+
+## Changelog
+
+Changes between `1.1.0` and this release: `1.2.0`
+
+- Update gatekeeper from `v3.1.0-beta.9` -> `v3.1.0`
+- Reenable High Availability
```
Comment (angelbarrera92) [inline docs/releases/unreleased.md:8]: reenable -> re-enable

File: docs/releases/unreleased.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+- The grafana dashboard is now part of this module
+- Namespaces `kube-system` and `gatekeeper-system` namespaces are exempted.
+
+## Upgrade path
+
+This version changes the labels for gatekeeper-controller-manager, so the usual `kustmize build | kubectl apply -f`
```
Comment (angelbarrera92) [inline docs/releases/unreleased.md:14]: ❤️

File: katalog/gatekeeper/monitoring/kustomization.yaml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+generatorOptions:
+  labels:
+    grafana-sighup-dashboard: default
+  disableNameSuffixHash: true
+
+configMapGenerator:
+  - name: grafana-dashboard-definitions-gatekeeper
+    files:
+      - gatekeeper-dashboard.json
```
Comment (angelbarrera92) [inline katalog/gatekeeper/monitoring/kustomization.yaml:17]: new line here

File: katalog/gatekeeper/rules/config/config.yml [hunk 4]
```diff
@@ -17,3 +17,6 @@ spec:
       - group: ""
         version: "v1"
         kind: "Pod"
+  match:
+    - excludedNamespaces: ["kube-system", "gatekeeper-system"]
+      processes: ["*"]
```
Comment (angelbarrera92) [inline katalog/gatekeeper/rules/config/config.yml:22]: new line here

File: katalog/gatekeeper/core/svc.yml [hunk 5]
```diff
@@ -16,3 +16,21 @@ spec:
...
+spec:
+  ports:
+    - port: 8888
+      targetPort: 8888
+      name: metrics
+  selector:
+    control-plane: audit-controller
+    gatekeeper.sh/operation: audit
+    gatekeeper.sh/system: "yes"
+---
```
Comment (angelbarrera92) [inline katalog/gatekeeper/core/svc.yml:36]: remove the last `---`

File: katalog/gatekeeper/core/deploy.yml [hunk 6]
```diff
@@ -36,7 +132,7 @@ spec:
               valueFrom:
                 fieldRef:
                   fieldPath: metadata.name
-          image: quay.io/open-policy-agent/gatekeeper:v3.1.0-beta.9
+          image: openpolicyagent/gatekeeper:v3.1.0
```
Comment (angelbarrera92) [inline katalog/gatekeeper/core/deploy.yml:135]: Nice movement from the opa team xD

File: katalog/gatekeeper/core/deploy.yml [hunk 7]
```diff
@@ -4,26 +4,122 @@ kind: Deployment
...
+        - name: POD_NAMESPACE
+          valueFrom:
+            fieldRef:
+              apiVersion: v1
+              fieldPath: metadata.namespace
+        - name: POD_NAME
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.name
+        image: openpolicyagent/gatekeeper:v3.1.0
```
Comment (angelbarrera92) [inline katalog/gatekeeper/core/deploy.yml:43]: Can you put the image tag in the kustomization.yaml file?
Something like:

```
image:
  - name: openpolicyagent/gatekeeper
    newTag: v3.1.0
```

Then in the deployment you don't need to specify the tag version

**Comment by ralgozino** [comment]
Fixed all the observations from @angelbarrera92 , could you please check the `.drone.yaml` file? I'm not sure the modification is OK. Thanks!

**Comment by angelbarrera92** [comment]
> Fixed all the observations from @angelbarrera92 , could you please check the `.drone.yaml` file? I'm not sure the modification is OK. Thanks!

I don't see anything strange, LGTM!

**Review by angelbarrera92 — APPROVED** [review]

**Pull request merged**